src/proxy/dns.rs

```diff
@@ -908,6 +916,38 @@ mod tests {
         }
     }
 
+    // TODO BML we might actually want to return both A and AAAA in this case, ultimately,
+    // and let the client deal with the mix.
+    // See https://datatracker.ietf.org/doc/html/rfc4038#section-3.2
+    // and https://github.com/istio/ztunnel/issues/582
+    #[tokio::test]
+    async fn ipv4_in_6_should_unwrap() {
+        let _guard = subscribe();
+        let fake_ips = vec![ip("2.2.2.2")];
+        let fake_wls = vec![xds_workload("client-fake", NS1, NW1, &fake_ips)];
+
+        // Create the DNS store.
+        let fake_state = new_proxy_state(&fake_wls, &[], &[]).state;
+        let forwarder = forwarder();
+        let store = DnsStore {
+            network: NW1.to_string(),
+            state: fake_state,
+            forwarder,
+        };
+
+        let ip4n6_client_ip = ip("::ffff:202:202");
+        let req = req(n("www.google.com"), ip4n6_client_ip, RecordType::A);
+        match store.lookup(&req).await {
+            Ok(_) => {}
+            Err(e) => {
+                if let Some(resp_code) = e.as_response_code() {
+                    assert_ne!(ResponseCode::ServFail, *resp_code);
```

Comment:
perhaps we want a stronger assertion here on a valid response code

Reply:
> perhaps we want a stronger assertion here on a valid response code

Yeah I was trying to minimally mock but I agree it can be better.